### PR TITLE
cat-log: fix file listing errors

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Include job-activity.log file if present even if listing is remote.
* List local files for submission-failed tasks.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/scripts/cat_log.py [hunk 1]
```diff
@@ -261,18 +261,18 @@ def view_log(
         # Print location even if the workflow does not exist yet.
         print(logpath)
         return 0
-    if not os.path.exists(logpath) and batchview_cmd is None:
```
Comment (oliver-sanders) [inline cylc/flow/scripts/cat_log.py:264]: The `logpath` contains the filename (defaults to `job.out`) even if it was not specified, even if we are listing a directory.

Consequently this caused failures when listing remote directories where the `job.out` file was not present.

**Review by wxtim — APPROVED** [review]
- [x] Read and understood
- [x] Tried (to break) it locally

**Review by MetRonnie — APPROVED** [review]

**Comment by wxtim** [comment]
Please hang fire until I've checked that this all works with the GUI.

**Review by MetRonnie — APPROVED** [review]
Awaiting tests

**Review by wxtim — APPROVED** [review]

**Pull request merged**